domain/src/main/java/org/oppia/domain/profile/ProfileManagementController.kt

```diff
@@ -97,15 +336,62 @@ class ProfileManagementController @Inject constructor(
     return ProfileId.newBuilder().setInternalId(currentProfileId).build()
   }
 
-  /**
-   * Sets the currentProfileId to the selected profile in ProfileChooserFragment.
-   * Checks to ensure that profileId corresponds to an existing profile.
-   *
-   * @param profileId the ID corresponding to the profile being set.
-   * @return a [LiveData] that indicates the success/failure of this set operation.
-   */
-  fun setCurrentProfileId(profileId: ProfileId): LiveData<AsyncResult<Any?>> {
-    currentProfileId = profileId.internalId
-    return MutableLiveData(AsyncResult.success<Any?>(null))
+  private fun isUniqueName(newName: String, profileDatabase: ProfileDatabase): Boolean {
+    val lowerCaseNewName = newName.toLowerCase()
+    profileDatabase.profilesMap.values.forEach {
+      if (it.name.toLowerCase(Locale.getDefault()) == lowerCaseNewName) {
+        return false
+      }
+    }
+    return true
+  }
+
+  private fun saveImageToInternalStorage(avatarImagePath: Uri, profileDir: File): String? {
+    val bitmap = MediaStore.Images.Media.getBitmap(context.contentResolver, avatarImagePath)
+    val imageFile = File(profileDir, "profile.png")
+    try {
+      FileOutputStream(imageFile).use { fos ->
+        rotateAndCompressBitmap(avatarImagePath, bitmap, 300)
+          .compress(Bitmap.CompressFormat.PNG, /* quality= */ 100, fos)
+      }
+    } catch (e: Exception) {
+      logger.e("ProfileManagementController", "Failed to store user submitted avatar image", e)
```

Comment:
Could we pipe this failure up to the data providers, instead? I think it'd be better if the UI is forced to handle it rather than hiding the error.

Or, is the intent that it's okay for this to fail?

Reply:
I think it's better to pipe up the failure. Updated to do so.